woocommerce/woocommerce-ios
# [Elevate] Add linked products banner

Part of #7374, #7377.

## Description

This PR adds a promo banner about cross-sells and upsells to product details view.


## Changes

- Adds new feature flag.
- Adds `LinkedProductsPromoCampaign` with correct analytics setup.
- Adds `FeatureAnnouncementCardView` with container row in `ProductForm`.
- Connects navigation from CTA banner button to linked products view.
- Updates `FeatureAnnouncementCard` with additional customization options.
- Updates `ProductForm` to display banner conditionally.

### Known limitations

Banner view is SwiftUI injected in UITableView, so there are some bits still WIP, mostly around animating it.

- Banner appears right away on product open, instead of animating in after save.
- Banner does not disappear on "X" tap (but it will save the interaction and never appear again).
- Banner changes its frame when scrolling the view

## Testing

1. Build and run the app in debug mode (to ensure the feature flag is enabled).
2. Open any product *with* linked products (upsells and/or cross-sells).
3. Confirm there is no promo banner.
4. Open product *without* any linked products.
5. Confirm promo banner appears.
6. Tap "Try it now" and confirm it navigates to linked products screen.
7. Add some products, tap done to return - confirm banner disappears.

## Screenshot

<img width=350 src=https://user-images.githubusercontent.com/3132438/182207904-51795bcb-ed56-4d67-b63a-0849f0d77042.png>

---
